controllers/directoryservice_controller.go

```diff
@@ -171,14 +176,31 @@ func (r *DirectoryServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	return requeue, nil
 }
 
-// SetupWithManager stuff
+// SetupWithManager function defines which objects the operator owns or watches then sends
+// reconcile requests to the Reconcile method based on those objects.
 func (r *DirectoryServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	// add this line
 	r.recorder = mgr.GetEventRecorderFor("DirectoryService")
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&directoryv1alpha1.DirectoryService{}).
 		Owns(&v1.Secret{}).        // Owns() triggers the Reconcile method for secrets we create
 		Owns(&apps.StatefulSet{}). // and statefulsets
+		Watches(				   // Watches ds-env-secrets for changes to service accounts passwords
+			&source.Kind{Type: &v1.Secret{ObjectMeta: metav1.ObjectMeta{
+				Name: "ds-env-secrets",
```

Comment:
This assumes the user always call these secrets by this name, right?

What if they want to use some other secret.  I think we need to watch the secrets specified by the spec.passwords,  and/or only the ones we generate.

Needs further discussion